virtcontainers/utils/utils_linux.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package utils
+
+/*
+#include <linux/vhost.h>
+
+#ifndef VHOST_VSOCK_SET_GUEST_CID
+  #define VHOST_VSOCK_SET_GUEST_CID _IOW(VHOST_VIRTIO, 0x60, __u64)
+#endif
+
+const int ioctl_VHOST_VSOCK_SET_GUEST_CID = VHOST_VSOCK_SET_GUEST_CID;
```

Comment:
Can we simply define `ioctl_VHOST_VSOCK_SET_GUEST_CID` on our own and have a comment above it referencing `#define VHOST_VSOCK_SET_GUEST_CID _IOW(VHOST_VIRTIO, 0x60, __u64) in <linux/vhost.h>
`? I still think we should avoid cgo as much as possible.

It is kernel ABI so it will be kept the same forever (or for a very very long time). So we do not need to worry about VHOST_VSOCK_SET_GUEST_CID being changed.

Reply:
Fair comment. We do have precedent as the runtime already makes use of cgo (see `cli/kata-check.go`), but, again, we could drop the dependency there if we adopted your approach.

Could you raise an issue on cgo and/or bring this issue up in the Architecture Committee meeting maybe so we can decide upon an approach (and document it ;) ?

Reply:
Fair comment. We do have precedent as the runtime does already make use of cgo (see `cli/kata-check.go`), but, again, we could drop the dependency there if we adopted your approach.

@bergwolf - Could you possibly raise an issue on the use of cgo so we can decide on our approach going forwards? We can discuss in the Architecture Committee meeting and then document the decision.